vita3k/renderer/src/gl/texture.cpp

```diff
@@ -467,25 +615,27 @@ void upload_bound_texture(const SceGxmTexture &gxm_texture, const MemState &mem)
             }
         }
 
-        const GLenum format = translate_format(base_format);
-        const GLenum type = translate_type(base_format);
-
-        glPixelStorei(GL_UNPACK_ROW_LENGTH, static_cast<GLint>(pixels_per_stride));
+        if (block_compressed) {
+            glPixelStorei(GL_UNPACK_ROW_LENGTH, 0);
 
-        if (need_decompress_and_unswizzle_on_cpu)
-            glTexSubImage2D(upload_type, mip_index, 0, 0, width, height, GL_RGBA, GL_UNSIGNED_BYTE, pixels);
+            const GLenum format = translate_format(base_format);
+            size_t compressed_size = renderer::texture::get_compressed_size(base_format, width, height);
+            glCompressedTexSubImage2D(upload_type, mip_index, 0, 0, width, height, format, static_cast<GLsizei>(compressed_size), pixels);
```

Comment:
Can you use glCompressedTexImage2D instead, Vita3K is still trying to be compatible with openGL 4.4 (the fact that we were using this function is a mistake).

Reply:
https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/glCompressedTexSubImage2D.xhtml
`glCompressedTexSubImage2D` is compatible even with OpenGL 2.0 while `glCompressedTextureSubImage2D` isn't

Reply:
Oops my bad, misread the doc